src/Sentry.AspNetCore/ScopeExtensions.cs

```diff
@@ -38,7 +38,10 @@ public static void Populate(this Scope scope, HttpContext context, SentryAspNetC
 
         if (options.SendDefaultPii && !scope.HasUser())
         {
-            var userFactory = context.RequestServices.GetService<ISentryUserFactory>();
+            // Although the compiler suggests context.RequestServices cannot be null, in practice it can be, so we
+            // have a null check here.
+            // See: https://github.com/getsentry/sentry-dotnet/issues/3395
+            var userFactory = context.RequestServices?.GetService<ISentryUserFactory>();
```

Comment:
Fair enough. But how do we end up in a state where we cannot locate an `ISentryuserFactory`?

Reply:
If we had a service collection reference, I think we'd find an `ISentryUserFactory` registered on it. I'm not sure what set of circumstances leads to `context.RequestServices` being null. That would require debugging some of the MS code, I think (or possibly the OpenTelemetry codebase). 

Since, in this case, the reference will only be null when nobody is logged in, it doesn't present any difficulty to us... we leave the user context empty in this case (exactly what we want).